aws/resource_aws_ssm_maintenance_window_task_test.go

```diff
@@ -51,6 +51,27 @@ func TestAccAWSSSMMaintenanceWindowTask_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSSSMMaintenanceWindowTask_noRole(t *testing.T) {
+	var task ssm.MaintenanceWindowTask
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_ssm_maintenance_window_task.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSSMMaintenanceWindowTaskDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSSSMMaintenanceWindowTaskNoRoleConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSSMMaintenanceWindowTaskExists(resourceName, &task),
+				),
+				ExpectNonEmptyPlan: true,
```

Comment:
Adding this type of handling in a test is indicative of a problem that practitioners would face in real world configurations:

```
=== CONT  TestAccAWSSSMMaintenanceWindowTask_noRole
    resource_aws_ssm_maintenance_window_task_test.go:59: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:


        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # aws_ssm_maintenance_window_task.test will be updated in-place
          ~ resource "aws_ssm_maintenance_window_task" "test" {
                id               = "b9865b4e-8afd-40da-89ec-4c90009f40c6"
                name             = "TestMaintenanceWindowTask"
              - service_role_arn = "arn:aws:iam::123456789012:role/aws-service-role/ssm.amazonaws.com/AWSServiceRoleForAmazonSSM" -> null
                # (7 unchanged attributes hidden)


                # (2 unchanged blocks hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
```

The solution in this case is to mark `service_role_arn` with `Computed: true` to allow Terraform to ignore the value being filled in when it is not configured. 👍